cmds/common_test.go

```diff
@@ -0,0 +1,56 @@
+package cmds
+
+import (
+	"testing"
+
+	"github.com/openshift/origin/pkg/project/api"
+)
+
+func TestGetProject(t *testing.T) {
+	p1 := api.Project{}
+	p2 := api.Project{}
+	p3 := api.Project{}
+	p4 := api.Project{}
+
+	p1.Name = "foo-che"
+	p2.Name = "bar-che"
+	p3.Name = "foo"
+	p4.Name = "moto"
+	// Pick the first one if when we have multiples projects and we are
+	// currently in an unrelated project
+	res := detectCurrentUserProject("moto", []api.Project{p1, p2, p3, p4})
+	if res != p3.Name {
+		t.Fatalf("%s != foo", res)
+	}
+
+	p1.Name = "foo-che"
+	p2.Name = "bar-che"
+	p3.Name = "foo"
+	p4.Name = "bar"
+	// Return the second project cause we are currently in there
+	res = detectCurrentUserProject("bar-che", []api.Project{p1, p2, p3, p4})
+	if res != "bar" {
+		t.Fatalf("%s != bar", res)
+	}
+
+	p1.Name = "foo-che"
+	p2.Name = "bar-che"
+	p3.Name = "foo"
+	p4.Name = "bar"
+	// Return bar because we are currently in the namespace bar-jenkins which has the same
+	// prefix)
+	res = detectCurrentUserProject("bar-jenkins", []api.Project{p1, p2, p3, p4})
+	if res != "bar" {
+		t.Fatalf("%s != %s", res, "bar")
+	}
+
+	// Return an error here, cause we have a foo-che but we don't have a parent
+	// project without prefix (i.e: foo)
```

Comment:
If we only find one project ending in `-che` such as `foo-che` and there's no "foo" then yes we should raise an error. I guess folks could have a `foo-che` and a `bar-che` and maybe `bar` exists but `foo` doesn't so there's no need to return an error, we could ignore `foo-che`